desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -109,13 +116,32 @@ def handle_on_link_shared(channel_id, message_ts, links):
     statement = doc_data['snippets'][0]['statement_raw'] if id_type == 'editor' else doc_data['statement_raw']
     dialect = doc_data['dialect'].capitalize() if id_type == 'editor' else doc.extra.capitalize()
     created_by = doc.owner.get_full_name() or doc.owner.username
-    
-    payload = _make_unfurl_payload(item['url'], statement, dialect, created_by)
+
+    user = rewrite_user(User.objects.get(username=doc.owner.username))
+    request = MockRequest(user=user)
+
+    result = query_result(request, doc_data)
```

Comment:
If this fails, return a 'query result expired or could not be found'?

Getting there it seems!